ownCloudAppShared/Client/File Lists/ClientQueryViewController.swift

```diff
@@ -476,10 +478,38 @@ extension ClientQueryViewController: UITableViewDropDelegate {
 						}
 					}
 				}
-			// Import Items from outside
 			} else {
-				guard let UTI = item.dragItem.itemProvider.registeredTypeIdentifiers.last else { return }
-				item.dragItem.itemProvider.loadFileRepresentation(forTypeIdentifier: UTI) { (url, _ error) in
+				// Import Items from outside
+				let typeIdentifiers = item.dragItem.itemProvider.registeredTypeIdentifiers
+				let preferredUTIs = [
+					kUTTypeImage,
+					kUTTypeMovie,
+					kUTTypePDF,
+					kUTTypeData
+				]
+				var useUTI : String?
+				var useIndex : Int = Int.max
+				
+				for typeIdentifier in typeIdentifiers {
+					if typeIdentifier != ItemDataUTI, !typeIdentifier.hasPrefix("dyn.") {
+						for preferredUTI in preferredUTIs {
```

Comment:
@felix-schwarz does it make sense to iterate over all preferredUTIs? Could also just `filter()` conforming one? Anyway first conforming UTI would win here.

Reply:
As in the comment above, it'd add an extra iteration and creating additional objects. An overhead that is avoided this way.